analyzers/src/SonarAnalyzer.Common/Rules/StringConcatenationInLoopBase.cs

```diff
@@ -154,25 +119,10 @@ private bool TryGetNearestLoop(SyntaxNode node, out SyntaxNode nearestLoop)
             return false;
         }
 
-        protected abstract bool IsInLoop(SyntaxNode node);
-
-        private bool IsDefinedInLoop(SyntaxNode expression, SyntaxNode nearestLoopForConcatenation,
-                SemanticModel semanticModel)
-        {
-            var symbol = semanticModel.GetSymbolInfo(expression).Symbol as ILocalSymbol;
-
-            var declaration = symbol?.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax();
-            if (declaration == null)
-            {
-                return false;
-            }
-
-            if (!TryGetNearestLoop(declaration, out var nearestLoop))
-            {
-                return false;
-            }
-
-            return nearestLoop == nearestLoopForConcatenation;
-        }
+        private bool IsDefinedInLoop(SyntaxNode expression, SyntaxNode nearestLoopForConcatenation, SemanticModel semanticModel) =>
+            semanticModel.GetSymbolInfo(expression).Symbol is var symbol
```

Comment:
Can be null:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.symbolinfo.symbol?view=roslyn-dotnet-4.6.0#microsoft-codeanalysis-symbolinfo-symbol